src/compiler/checker.ts

```diff
@@ -263,6 +263,30 @@ namespace ts {
         Uncapitalize: IntrinsicTypeKind.Uncapitalize
     }));
 
+    enum TypeChecks {
+        Ok,
+        HasDiagnostic
+    }
+
+    interface CheckerResultOk {
+        result: TypeChecks.Ok
+    }
+
+    interface CheckerResultHasDiagnostic {
+        result: TypeChecks.HasDiagnostic,
+        message: DiagnosticMessage;
+        args:
+            | never[]

```

Comment:
Does `[]` work here too?

Reply:
not if we use `emptyArray`, which has type `never[]`, but I might be missing something. anyways, going to get rid of this so it doesn't matter for now